pkg/volume/plugins.go

```diff
@@ -317,6 +318,23 @@ func NewSpecFromPersistentVolume(pv *api.PersistentVolume, readOnly bool) *Spec
 	}
 }
 
+// NewSpecFromVolumeWithNS creates an Spec from an api.Volume and a namespace
+func NewSpecFromVolumeWithNS(vs *api.Volume, nameSpace string) *Spec {
```

Comment:
To pass namespace to `volumeSpec`, we either refactor 1) `NewSpecFromVolume` to add a second parameter `namespace` or 2) create a new constructor (as in here)

As I stated in the PR, I tried both ways. The sad story of refactoring `NewSecFromVolume` is that it creates a massive refactoring to many innocent files. Thus I believe a new constructor is a cleaner way for the job: it is only used in the place where namespace is needed.